pkg/registry/secrets.go

```diff
@@ -26,78 +26,58 @@ const (
 	RegistryS3SecretReadyConditionType = "RegistryS3SecretReady"
 )
 
-func EnsureSecrets(ctx context.Context, in *clusterv1beta1.Installation, metadata *ectypes.ReleaseMetadata, cli client.Client) error {
-	if in == nil || !in.Spec.AirGap || !in.Spec.HighAvailability {
-		return nil
-	}
-
+func EnsureSecrets(ctx context.Context, in *clusterv1beta1.Installation, cli client.Client, seaweedfsExt, registryExt k0sv1beta1.HelmExtensions) error {
 	log := ctrl.LoggerFrom(ctx)
 
-	config, op, err := ensureSeaweedfsS3Secret(ctx, in, metadata, cli)
+	sfsConfig, op, err := ensureSeaweedfsS3Secret(ctx, in, cli, seaweedfsExt)
 	if err != nil {
-		in.Status.SetCondition(metav1.Condition{
-			Type:               SeaweedfsS3SecretReadyConditionType,
-			Status:             metav1.ConditionFalse,
-			Reason:             "SecretFailed",
-			Message:            err.Error(),
-			ObservedGeneration: in.Generation,
-		})
 		if errors.IsFatalError(err) {
+			in.Status.SetCondition(getSeaweedfsS3SecretReadyCondition(in, metav1.ConditionFalse, "ReleaseMetadataFailed", err.Error()))
```

Comment:
why are you calling this "ReleaseMetadataFailed"? is there anything to do with metadata here?

Reply:
I guess the only fatal errors you're throwing are metadata related, but to me this says "set the installation status a level deeper in the code"